src/lr_apply.rs

```diff
@@ -41,162 +41,152 @@ unsafe fn lr_stripe<BD: BitDepth>(
     mut edges: LrEdgeFlags,
 ) {
     let seq_hdr = &***(*f).seq_hdr.as_ref().unwrap();
-    let dsp: *const Rav1dDSPContext = (*f).dsp;
+    let dsp: &Rav1dDSPContext = &*(*f).dsp;
     let chroma = (plane != 0) as c_int;
-    let ss_ver = chroma
-        & ((*f).sr_cur.p.p.layout as c_uint == Rav1dPixelLayout::I420 as c_int as c_uint) as c_int;
+    let ss_ver = chroma & ((*f).sr_cur.p.p.layout == Rav1dPixelLayout::I420) as c_int;
     let stride: ptrdiff_t = (*f).sr_cur.p.stride[chroma as usize];
-    let sby =
-        y + (if y != 0 {
-            (8 as c_int) << ss_ver
-        } else {
-            0 as c_int
-        }) >> 6 - ss_ver + seq_hdr.sb128;
+    let sby = y + (if y != 0 { 8 << ss_ver } else { 0 }) >> 6 - ss_ver + seq_hdr.sb128;
     let have_tt = (c.tc.len() > 1) as c_int;
-    let mut lpf: *const BD::Pixel = ((*f).lf.lr_lpf_line[plane as usize] as *mut BD::Pixel)
-        .offset(
-            (have_tt * (sby * ((4 as c_int) << seq_hdr.sb128) - 4)) as isize
-                * BD::pxstride(stride as usize) as isize,
-        )
-        .offset(x as isize);
+    // `lpf_offset` can be negative when `have_tt` is 1 and `sby` is 0.
+    // To avoid defining a slice with elements outside the allocated memory area, a
+    // variable `lpf_ptr_offset` is added to keep track of a negative offset that
+    // has to be added when casting the slice to a raw pointer.
+    let lpf_offset = (have_tt * (sby * (4 << seq_hdr.sb128) - 4)) as isize
+        * BD::pxstride(stride as usize) as isize
+        + x as isize;
+    let mut lpf_ptr_offset = if lpf_offset < 0 {
+        -4 * BD::pxstride(stride as usize) as isize
+    } else {
+        0
+    };
```

Comment:
`lpf_ptr_offset` is always unsigned, right?  `lpf_offset < 0` only when `stride < 0`, so `lpf_ptr_offset` is positive in that case.  Could you make it a `usize` then?

Reply:
`lpf_offset` can be `<0` even if `stride > 0`, in which case `lpf_ptr_offset` is negative too